kernel/packages/shared/meta/index.ts

```diff
@@ -20,3 +23,20 @@ export async function ensureMetaConfigurationInitialized(): Promise<void> {
     })
   })
 }
+
+export async function waitForMessageOfTheDay(): Promise<MessageOfTheDayConfig | null> {
+  const store: Store<RootState> = globalThis.globalStore
+  const worldConfig = store.getState().meta.config.world
+  if (worldConfig && worldConfig.messageOfTheDayInit) {
+    return Promise.resolve(worldConfig.messageOfTheDay || null)
+  }
+  return new Promise<MessageOfTheDayConfig | null>((resolve) => {
+    const unsubscribe = globalThis.globalStore.subscribe(() => {
```

Comment:
What happens if the request for message of the day fails? Does this stay blocked forever?